src/grading-settings/grading-scale/GradingScale.jsx

```diff
@@ -55,6 +56,15 @@ const GradingScale = ({
 
   const addNewGradingSegment = () => {
     setGradingSegments(prevSegments => {
+      if (prevSegments.length >= 5) {
+        const segSize = MAXIMUM_SCALE_LENGTH / (prevSegments.length + 1);
+        return Array.from({ length: prevSegments.length + 1 }).map((_, i) => (
+          {
+            current: 100 - i * segSize,
+            previous: 100 - (i + 1) * segSize,
+          }
+        ));
+      }
```

Comment:
If the number of segments is more than 5 this will switch to just splitting the segments evenly.

Reply:
because of the return statement here the "showSavePrompt" below is not triggered I am investigating a bit more.

Reply:
@farhaanbukhsh  Thanks for identifying an fixing this! I'll incorporate your fix into my pr.